### PR TITLE
chore: update to rust 1.78.0

### PR DESCRIPTION
**Pull request by helio-frota**

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.76.0"
+channel = "1.78.0"
```
Comment (bobmcwhirter) [inline rust-toolchain.toml:2]: nitpick: could you limit this PR to exactly this line, and not the other changes?

**Review by bobmcwhirter — CHANGES_REQUESTED** [review]

**Review by bobmcwhirter — APPROVED** [review]

**Comment by bobmcwhirter** [comment]
Ah... now I understand why you had those other changes... new clippy lints, yeah?

**Comment by helio-frota** [comment]
yeah , lol

File: common/infrastructure/src/app/http.rs [hunk 2]
```diff
@@ -396,10 +396,8 @@ impl HttpServerBuilder {
             .build();
 
         self.metrics_factory = Some(Arc::new(move || {
-            metrics
-                .as_ref()
-                .map(|r| r.clone())
-                .map_err(|err| anyhow!("{err}"))
+            let m = metrics.as_ref();
+            m.cloned().map_err(|err| anyhow!("{err}"))
```
Comment (jcrossley3) [inline common/infrastructure/src/app/http.rs:400]: I don't think you need the local `m`. You could just `metrics.as_ref().cloned().map_err(|err| anyhow!("{err}"))`

Comment (helio-frota) [reply common/infrastructure/src/app/http.rs:400]: let me try :+1:

**Pull request merged**